docs/AUTHORING.md

```diff
@@ -0,0 +1,166 @@
+# Authoring Plugins
+
+This document gives guidance for authoring plugins.
+
+SPIRE plugins implement one and only one plugin _type_ (e.g. KeyManager). They
+also implement zero or more services.
+
+## Configuration
+
+Most plugins require some form of configuration. SPIRE supports passing plugin
+configuration data to plugins when they are loaded. This configuration data is
+provided to SPIRE in the `plugin_data` section of the plugin declaration in the
+server or agent configuration, e.g.:
+
+```
+plugins {
+    UpstreamAuthority "disk" {
+        plugin_data {
+            key_file_path = "some.key"
+            cert_file_path = "some.crt"
+        }
+    }
+```
+
+In order to receive this configuration data, as well as other core
+configurables, a plugin implements the [Config](proto/spire/service/common/config) service:
```

Comment:
The link will not work, as it's relative to the docs dir.

Reply:
Fixed.